src/client/src/apps/SimpleLauncher/Launcher.tsx

```diff
@@ -1,119 +1,135 @@
+  const searchInputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    getCurrentWindowBounds().then(setInitialBounds)
+  }, [])
+
+  const showSearch = useCallback(
+    () => {
+      if (isSearchVisible) {
+        searchInputRef.current && searchInputRef.current.focus({ preventScroll: true })
+      }
+      setIsSearchVisible(true)
+    },
+    [isSearchVisible]
+  );
+
+  // if search is made visible - focus on it
+  useEffect(
+    () => {
+      if (isSearchVisible) {
+        searchInputRef.current && searchInputRef.current.focus({ preventScroll: true })
+      }
+    },
+    [isSearchVisible]
+  )
 
-const IconTitle = styled.span`
-  position: absolute;
-  bottom: 2px;
-  right: 0;
-  left: 0;
-  text-align: center;
-  font-size: 9px;
-  font-family: Lato;
-  color: transparent;
-  transition: color 0.3s ease;
+  // hide search if Escape is pressed
+  useEffect(
+    () => {
+      const onKeyDown = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          setIsSearchVisible(false)
+        }
+      }
+      document.addEventListener('keydown', onKeyDown)
+      return () => document.removeEventListener('keydown', onKeyDown)
+    }, []
+  )
 
-  /* avoids text highlighting on icon titles */
-  user-select: none;
-`
+  const handleSearchStateChange = useCallback(
+    (state: SearchState) => setIsSearchBusy(state.typing || state.loading),
+    []
+  )
 
-const IconContainer = styled.div`
-  height: 100%;
-  width: 100%;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  flex-direction: column;
-  &:hover {
-    span {
-      color: ${({ theme }) => theme.core.textColor};
-    }
-  }
-`
+  const handleSearchSizeChange = useCallback(
+    (contentRect: ContentRect) => {
+      if (!initialBounds) {
+        return
+      }
+      animateCurrentWindowSize({
+        ...initialBounds,
+        height: initialBounds.height + (contentRect.bounds ? contentRect.bounds.height : 0)
+      })
+    },
+    [initialBounds]
+  )
 
-const ButtonContainer = styled(IconContainer)`
-  ${rules.appRegionNoDrag};
-`
-const ThemeSwitchContainer = styled(ButtonContainer)`
-  width: 35%;
-`
+  return (
+    <RootContainer>
+      <LauncherGlobalStyle/>
 
-const LogoContainer = styled(IconContainer)`
-  width: 50%;
-  background-color: ${({ theme }) => theme.core.lightBackground};
-  .svg-icon {
-    fill: ${({ theme }) => theme.core.textColor};
-  }
-  ${rules.appRegionDrag};
-`
+      <HorizontalContainer>
+        <LogoContainer>
+          <AdaptiveLoader size={24} speed={isSearchBusy ? 0.8 : 0} seperation={1.5} type="secondary"/>
+        </LogoContainer>
+        {/* TODO: enable in production when we are ready */}
+        {process.env.NODE_ENV === 'development' ? <SearchButton onClick={showSearch}/> : null}
+        <LauncherApps/>
+        <LauncherExit/>
+        <ThemeSwitchContainer>
+          <ThemeStorageSwitch/>
+        </ThemeSwitchContainer>
+      </HorizontalContainer>
+
+      <Measure
```

Comment:
`<Measure>` is third-party thing that helps us to dynamically react to size changes